client/src/main/java/org/apache/nemo/client/ClientEndpoint.java

```diff
@@ -164,7 +164,8 @@ public final Enum waitUntilJobFinish(final long timeout,
         return stateTranslator.translateState(driverEndpoint.get().
             waitUntilFinish(timeout - unit.convert(consumedTime, TimeUnit.NANOSECONDS), unit));
       } else {
-        return PlanState.State.READY;
+        // This default value needs double checking.
+        return stateTranslator.translateState(PlanState.State.READY);
```

Comment:
Could you please indicate why we are changing the return type here please?

As far as I could infer, `stateTranslator.translateState(PlanState.State.READY)` equals `RUNNING` defined at `PipelineResult.java`while `PlanState.State.READY` equals a `READY` defined at `PlanState.java`

I do not understand why we are changing here. And where is this function used please?

Reply:
This is because driverEndpoint.get() == null indicates the program is running? Is  this correct @wonook ? and `driverEndpoint.get() == null` guarantees running state?

Reply:
@wonook  
@taegeonum and me think that `ClientEndpoint` and `DriverEndpoint` are not used anymore. We presume these are just legacy code we forgot to remove. 
What do you think?

Reply:
@johnyangk Any insights on the ClientEndpoint? I think it'd be better to make a separate issue to clean up the code, if it is not used anymore.

Reply:
By the way this change was made to sync the return type with all other types. This was the only case where it was PlanState.State, instead of Beam's PipelineResult.State

Reply:
@johnyangk 
Would you confirm if ClientEndpoint and DriverEndpoint are still used or not?
What is the intention of the two classes?